File: sample_programs/pirate.cpp
// Dr. Mohan
// Demonstration program for lab 5. This program is used in
// conjunction with the following files:
//       lab5demo.cpp
//       pirate.h
//       pirate.cpp
//       parrot.h
//       parrot.cpp

#include <string>
#include "pirate.h"

using namespace std;

Pirate::Pirate(string n) { Pirate::name = n; }

string Pirate::getName() { return Pirate::name; }

string Pirate::speak() { return "Arrrr!"; }

File: sample_programs/pirate.cs
// Dr. Mohan
// Lab 5 demonstration program
//
// See files "parrot.cs" and "lab5demo.cs" for the
// other parts of this program.

using System;

namespace PirateNS
{
   // A Pirate has a name and always says "Arrrr!"
   class Pirate {
      private string name;
   
      // Constructor:
      public Pirate(string n) {
         name = n;
      }
   
      public string getName() { return name; }
   
      public string speak() { return "Arrrr!"; }
   }
}
